### ISSUE TITLE
Duplicate pre tags produced

### ISSUE THREAD

**Issue by alampros** [bebraw/remark-react-lowlight#3]
Fenced code blocks are producing duplicate (nested) `<pre>` tags.

## Markdown:

     ```html
     <i class="icon icon-check" />
     ```

## Output

```html
<pre>
  <pre class="lowlight">
     <code class="hljs html">
     ...
     </code>
  </pre>
</pre>
```

## Expected Output

```html
<pre class="lowlight">
   <code class="hljs html">
   ...
   </code>
</pre>
```

**Comment by bebraw** [comment]
Thanks for the report. Do you want to PR a fix?

**Comment by alampros** [comment]
I don't think I have enough expertise with remark to fix it. If it would help, I could take a stab at a PR for a test case to reproduce the issue.

**Comment by bebraw** [comment]
Sure. A test case would be great. Easy enough to fix then. Thanks.

**Comment by rkaneko** [comment]
@alampros 

I recognized this issue. However, I couldn't think of a good solution.
Previously, I discussed hast API with woorms. [logs](https://twitter.com/__rkaneko/status/791478720410234880)

As a result of the discussion, I concluded it it difficult to support both the inline code and the block level code in the remark-react-lowlight.

This is because

- We couldn't confirm the code block is inline or not to see the props bound from remark-react.
- To confirm the code is inline or not, we need check the parent is the pre element or not. However, we need some flags to manage the state of parsing while converting hast to markdown.

I think it is not a good design to use some flags.
As common ground of remark-react-lowlight library, it is better to support only block level code.

This is one of PoC code examples.

- https://github.com/rkaneko/ambient/blob/1ca4d541ad9441aad7c4f488ac313a8ccabcda2c/src/components/code-block/code-block.js
- https://github.com/rkaneko/ambient/blob/1ca4d541ad9441aad7c4f488ac313a8ccabcda2c/src/js/application/md-to-react.js#L9-L16

What do you think about this issue? @bebraw

**Comment by bebraw** [comment]
Wouldn't it be possible to tell these cases part based on the fact whether or not `language` exists? That's not set for inline code for sure. I think this might cover most of the need and allow at least some level of checking.

**Comment by rkaneko** [comment]
OK. This library's giving up supporting the block level code without language, right?

If you agree, I'm going to push the feature branch!

**Comment by bebraw** [comment]
> OK. This library's giving up supporting the block level code without language, right?

Yeah.

Interestingly we already have `language ? <Lowlight {...props} /> : <code>{ value }</code>`. That `pre` issue might stem from `Lowlight` itself.

**Comment by rkaneko** [comment]
Hmm... I need much time.

I tried

```diff
-    const props = { language, value };
-    return language ? <Lowlight {...props} /> : <code>{ value }</code>;
+    const inline = !language;
+    const props = { language, value, inline };
+    return <Lowlight {...props} />;
```

However, I’ｍ in some troubles.

- Rendering a block-level code without defined language is different from what I expected. (Maybe escaping algorithm of some library is complicated. But I haven't search in detail yet.)
- Rendering a inline code is different from what I expected. (Some text will disappear by escaping.)

example

```markdown
   ```js
   function blah(arg1) {
     console.log(arg1);
   }
    ```
    ```
   public class App {
     public void static main(String[] args) {
       System.out.println("Hello world!");
     }
   }
    ```

- `hoge`
```

rendering result is

![screenshot from 2016-12-06 01-09-51](https://cloud.githubusercontent.com/assets/2318465/20892249/bef4b89a-bb50-11e6-8d29-2cc9984e5822.png)


So, I need more time to investgate libraries remark-react-lowlight depends on.

**Comment by bebraw** [comment]
No worries. 👍

**Comment by sugarshin** [comment]
Hi,
Has this issue been solved? I still have that problem.

**Comment by rkaneko** [comment]
@sugarshin 

I'm sorry this issue has not be solved yet.
I don't  get how to solve this problem.

**Comment by sugarshin** [comment]
@rkaneko Thank you for reply.

Perhaps I think that [this](https://github.com/bebraw/remark-react-lowlight/issues/3#issuecomment-264899784) solution will solve it. I tried it locally on my own, but it looks like it does not seem to be problematic.

**Comment by sugarshin** [comment]
I think everything will be settled by doing this.

```diff
-    const props = { language, value };
+    const props = { language, value, inline: true };
```

**Comment by bebraw** [comment]
@sugarshin Can you PR that? Be sure to document the change well. 👍

**Comment by sugarshin** [comment]
@bebraw I've already done. 🙏

**Comment by bebraw** [comment]
This should be fixed in `0.4.0`. I also upgraded the project setup while at it so it's easier to develop.
